src/renderer/src/lib.rs

```diff
@@ -184,31 +184,43 @@ pub struct Fragment<R: gfx::Resources> {
     pub ka: Texture<R>,
     /// diffuse color
     pub kd: Texture<R>,
+    /// specular color
+    pub ks: Texture<R>,
+    /// specular exponent
+    pub ns: f32,
 }
 
-/// A basic light
+
+/// Represents a point light.
 #[derive(Copy, Clone)]
-pub struct Light {
+pub struct PointLight {
     /// The XYZ coordinate of the light
     pub center: [f32; 3],
-    /// How big the light is radius, lighting
-    /// passes are not required to render the light
-    /// beyond this radius
-    pub radius: f32,
 
     /// The color of light emitted
     pub color: [f32; 4],
-    /// constant, propagation means no falloff of the light
-    /// emission from distance. Useful for the sun.
-    pub propagation_constant: f32,
-    /// linear level drops
-    pub propagation_linear: f32,
-    /// cubic light level drop
-    pub propagation_r_square: f32,
+
+    /// Represents constant, linear, and quadratic (kc, kl, and kq)
+    /// factors for light attenuation calculations. See these links
+    /// for some more information on how to choose good values for
+    /// these factors:
```

Comment:
Should also provide a function to generate such useful values for the user e.g from a light radius